Packs/SignalSciences/Integrations/SignalSciences/SignalSciences.py

```diff
@@ -445,42 +445,50 @@ def gen_entries_data_for_update_list_request(entries_list, method):
 
 
 def gen_context_for_add_to_whitelist_or_blacklist(response_data):
-    ip_context = {}
-    ip_context['ID'] = response_data.get('id', '')
-    ip_context['Note'] = response_data.get('note', '')
-    ip_context['Source'] = response_data.get('source', '')
-    ip_context['CreatedBy'] = response_data.get('createdBy', '')
-    ip_context['CreatedDate'] = response_data.get('created', '')
-    ip_context['ExpiryDate'] = response_data.get('expires', '')
-    return ip_context
-
-
-def generate_whitelist_or_blacklist_ip_context(response_data):
-    ips_contexts = []
-    for ip_data in response_data:
-        cur_ip_context = gen_context_for_add_to_whitelist_or_blacklist(ip_data)
-        ips_contexts.append(cur_ip_context)
-    return ips_contexts
+    full_data = []
+    for data in response_data:
+        full_data.append({
+            'ID': data.get('id', ''),
+            'Note': data.get('note', ''),
+            'Source': data.get('source', ''),
+            'CreatedBy': data.get('createdBy', ''),
+            'CreatedDate': data.get('created', ''),
+            'ExpiryDate': data.get('expires', '')
+        })
+    return full_data
 
 
 def gen_human_readable_for_add_to_whitelist_or_blacklist(ip_context):
-    human_readable = {}
-    human_readable['Note'] = ip_context['Note']
-    human_readable['Source'] = ip_context['Source']
-    human_readable['Expiration date'] = ip_context['ExpiryDate'] if ip_context['ExpiryDate'] else "Not Set"
+    human_readable = []
+    for context in ip_context:
+        human_readable.append({
+            'Note': context['Note'],
+            'Source': context['Source'],
+            'Expiration date': context['ExpiryDate'] if context['ExpiryDate'] else "Not Set"
+        })
     return human_readable
 
 
 def add_ip_to_whitelist_or_blacklist(url, ip, note, expires=None):
-    check_ip_is_valid(ip)
-    data = {
-        'source': ip,
-        'note': note
-    }
-    if expires is not None:
-        data['expires'] = expires
-    res = http_request('PUT', url, data=data)
-    return res
+    res_list = []
+    error_list = []
+    for single_ip in argToList(ip):
+        try:
+            check_ip_is_valid(single_ip)
+            data = {
+                'source': single_ip,
+                'note': note
+            }
+            if expires is not None:
+                data['expires'] = expires
+            res_list.append(http_request('PUT', url, data=data))
+        except SystemExit:
+            # handle exceptions in return_error
+            pass
```

Comment:
no need to check that, the first failure should stop the command (no need to continue to the rest of the least).

relevant for all.

Reply:
i have fixed it in the other integrations.
but in this one specific it’s hard to fix it.